### PR TITLE
Feat/react

### PR DESCRIPTION
**Pull request by jfalxa**
Hi, i'll explain you with my voice if you plz.

### CODE REVIEW

File: src/Editor.js [hunk 1]
```diff
@@ -1,39 +1,66 @@
...
 			value: this.props.content,
-			mode:  "javascript"
+			mode: 'javascript'
 		})
 
 		this.codeMirror.on('change', this.handleCodeMirrorChange)
 	}
 
+	componentDidUpdate(prevProps) {
+		const { '*': noteId, content, onLoad, onReset } = this.props
```
Comment (hugurp) [inline src/Editor.js]: Isn't it kind of _super_weird_™  that `*` is a variable, and we assign it after to `noteId` etc.

File: src/App.js [hunk 2]
```diff
@@ -58,54 +69,38 @@ class App extends Component {
...
+				<Nav />
 
 				<Router>
-
 					<Editor
-						default
-						path='/:id'
-						content={ content }
-						onChange={ this.handleContentChange } />
+						path="/*"
```
[the remark this answers is not in the document]
Comment (jfalxa) [reply src/App.js]: https://github.com/reach/router/issues/164

File: src/App.js [hunk 3]
```diff
@@ -58,54 +69,38 @@ class App extends Component {
...
+						content={content}
+						onChange={this.handleContentChange}
+						onLoad={this.loadNote}
```
[the remark this answers is not in the document]
Comment (jfalxa) [reply src/App.js]: App.js exposes all the methods needed for manipulating notes, you pass them to the right children so they use them only when needed.

File: src/Editor.js (hunk 1 above)
Comment (jfalxa) [reply src/Editor.js]: no

**Comment by hugurp** [comment]
Thanks a million for this PR @jfalxa !

I'm merging all open PRs today, so the site can be put online; I'll test everything manually over the next few days/weeks/month and increment new smallish features to make the UX inkredible™

peace

**Pull request merged**